internal/controllers/clusterrequest_controller.go

```diff
@@ -1417,23 +1485,36 @@ func (t *clusterRequestReconcilerTask) updateClusterInstance(ctx context.Context
 	return true
 }
 
-// waitForHardwareData waits for the NodePool to be provisioned and update BMC details in ClusterInstance.
-func (t *clusterRequestReconcilerTask) waitForHardwareData(ctx context.Context,
-	clusterInstance *unstructured.Unstructured, nodePool *hwv1alpha1.NodePool) bool {
+// Updates the Node status with the hostname after BMC information has been assigned.
+func (t *clusterRequestReconcilerTask) updateNodeStatusWithHostname(ctx context.Context, nodeName, hostname, namespace string) bool {
+	node := &hwv1alpha1.Node{}
+	exists, err := utils.DoesK8SResourceExist(ctx, t.client, nodeName, namespace, node)
+	if err != nil || !exists {
+		t.logger.ErrorContext(
+			ctx,
+			"Failed to get the Node object for updating hostname",
+			slog.String("name", nodeName),
+			slog.String("namespace", namespace),
+			slog.String("error", err.Error()),
+			slog.Bool("exists", exists),
+		)
+		return false
```

Comment:
Same comment in the function as the others related to returning of the error.

Reply:
The error is already logged in the proper context prior to returning, so returning the error to the caller wouldn't provide additional benefit.